src/EditorFeatures/Test/EmbeddedLanguages/StackFrame/StackFrameParserTests.cs

```diff
@@ -0,0 +1,366 @@
+                    QualifiedName("M.M"),
+                    argumentList: EmptyParams),
+
+                fileInformation: FileInformation(
+                    Path(@"C:\folder\m.cs"),
+                    ColonToken,
+                    Line(1).With(trailingTrivia: CreateTriviaArray("[trailingtrivia]"))),
+
+                eolTokenOpt: EOLToken
+            );
+
+        [Fact]
+        public void TestFileInformation_InvalidDirectory()
+            => Verify(@"M.M() in <\m.cs", expectFailure: true);
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("lkasjdlkfjalskdfj")]
+        [InlineData("\n")]
+        [InlineData("at ")]
+        [InlineData(@"at M()")] // Method with no class is invalid
+        [InlineData(@"at M.1c()")] // Invalid start character for identifier
+        [InlineData(@"at 1M.C()")]
+        [InlineData(@"at M.C(string& s)")] // "string&" represents a reference (ref, out) and is not supported yet
+        [InlineData(@"at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__139`1.MoveNext()")] // Generated/Inline methods are not supported yet
+        [InlineData(@"at M(")] // Missing closing paren
+        [InlineData(@"at M)")] // MIssing open paren
+        [InlineData(@"at M.M[T>(T t)")] // Mismatched generic opening/close
+        [InlineData(@"at M.M<T](T t)")] // Mismatched generic opening/close
+        [InlineData(@"at M.M(string[ s)")] // Opening array bracket no close
+        [InlineData(@"at M.M(string] s)")] // Close only array bracket
+        [InlineData(@"at M.M(string[][][ s)")]
+        [InlineData(@"at M.M(string[[]] s)")]
+        [InlineData(@"at M.N`.P()")] // Missing numeric for arity 
+        [InlineData(@"at M.N`9N.P()")] // Invalid character after arity
+        [InlineData("M.N.P.()")] // Trailing . with no identifier before arguments
+        [InlineData("M.N(X.Y. x)")] // Trailing . in argument type
+        [InlineData("M.N[T.Y]()")] // Generic type arguments should not be qualified types
+        [InlineData("M.N(X.Y x.y)")] // argument names should not be qualified
+        public void TestInvalidInputs(string input)
+            => Verify(input, expectFailure: true);
+
+        [Theory]
+        [InlineData("at ")]
+        [InlineData("in ")]
+        [InlineData("line ")]
+        public void TestKeywordsAsIdentifiers(string keyword)
+            => Verify(@$"MyNamespace.MyType.MyMethod[{keyword}]({keyword} {keyword})",
+                methodDeclaration: MethodDeclaration(
+                    QualifiedName("MyNamespace.MyType.MyMethod"),
+                    typeArguments: TypeArgumentList(TypeArgument(IdentifierToken(keyword.Trim(), trailingTrivia: SpaceTrivia()))),
+                    argumentList: ParameterList(
+                        Parameter(
+                            Identifier(keyword.Trim()),
+                            IdentifierToken(keyword.Trim(), leadingTrivia: SpaceTrivia(2), trailingTrivia: SpaceTrivia()))))
+                );
+    }
```

Comment:
i'm not sure if you already did it, but:

1. have your tests validate you don't crash on any prefix/suffix string of your test strings.  
2. for every node type try to test a few edge cases to make sure you don't crash.

Reply:
for the edge cases.  think about empty tokens, or an extra intermediary token.  so things like `A.B.(params)`

Reply:
> have your tests validate you don't crash on any prefix/suffix string of your test strings

Added as inputs to `TtestIdentifierNames` 

> for every node type try to test a few edge cases to make sure you don't crash

I tried to do this with various input into `TestInvalidInputs`. Let me know if you can think of explicit things I missed